### PR TITLE
feat: support setting a discard type of reporter

### PR DESCRIPTION
**Pull request by LinuxSuRen**
As mentioned in the title, users could discard the report data via environment variable setting.

### CODE REVIEW

File: tools/go-agent/Makefile [hunk 1]
```diff
@@ -51,4 +51,4 @@ build: linux darwin windows
 .PHONY: $(PLATFORMS)
 $(PLATFORMS):
 	mkdir -p $(OUT_DIR)
-	GOOS=$(os) GOARCH=$(ARCH) $(GO_BUILD) $(GO_BUILD_FLAGS) -ldflags "$(GO_BUILD_LDFLAGS)" -o $(OUT_DIR)/$(BINARY)-$(VERSION)-$(os)-$(ARCH) ./cmd
+	CGO_ENABLED=0 GOOS=$(os) GOARCH=$(ARCH) $(GO_BUILD) $(GO_BUILD_FLAGS) -ldflags "$(GO_BUILD_LDFLAGS)" -o $(OUT_DIR)/$(BINARY)-$(VERSION)-$(os)-$(ARCH) ./cmd
```
Comment (LinuxSuRen) [inline tools/go-agent/Makefile:54]: Will occur the `GLIC` relevant issues.

**Comment by mrproliu** [comment]
Can you provide more scenarios to illustrate when users need to use it? If they don't need to collect data and can perform operations during compilation, it will result in better performance.

**Comment by LinuxSuRen** [comment]
There are two important reason.

* Considering SkyWalking is not a MUST component all the time. I don't expect that it prints error logs when no SkyWalking installed.
* Maintaining the double number of binary files is an annoying thing. Let's assume one project has different binary files against Windows, Linux, macOS. If there is no such switch to turn off reporting, then users need to publish 6 packages everytime instead of 3. Even wrose if arm64 is required, users might need to mtaintain 12 packages.

**Review by mrproliu — APPROVED** [review]

**Review by mrproliu — COMMENTED** [review]
Then, could you please add this feature in the [CHANGE.md](https://github.com/apache/skywalking-go/blob/main/CHANGES.md)

**Review by mrproliu — APPROVED** [review]
Thanks!

**Pull request merged**